packages/ketcher-react/src/script/ui/component/form/input.tsx

```diff
@@ -158,19 +183,29 @@ function FieldSet({
 }
 
 FieldSet.val = function (ev, schema) {
-  const input = ev.target
+  const input = ev.target as HTMLInputElement
+
   if (ev.target.tagName !== 'INPUT') {
     ev.stopPropagation()
     return undefined
   }
+
   // Hm.. looks like premature optimization
   //      should we inline this?
-  const fieldset = input.parentNode.parentNode.parentNode
-  const result = [].reduce.call(
-    fieldset.querySelectorAll('input'),
-    (res, inp, i) => (!inp.checked ? res : [enumSchema(schema, i), ...res]),
-    []
-  )
+
+  const fieldset = input?.parentNode?.parentNode?.parentNode
+  const inputCollection = fieldset?.querySelectorAll('input')
+  let result
+
+  if (inputCollection && inputCollection.length) {
+    result = Array.from(inputCollection).reduce(
+      (res, inp: HTMLInputElement, i) =>
+        !inp.checked ? res : [enumSchema(schema, i), ...res],
+
+      [] as HTMLInputElement[]
```

Comment:
Not sure it is a good idea - inside this function, we will need to check if it is an array of inputs or options - so that using  'checked' or 'selected' param, and cast types somehow, I think it will look a bit ugly and wouldn't make things clearer.